waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/KeyController.kt

```diff
@@ -230,7 +230,22 @@ fun Application.keys() = walletRoute {
                 val keyId = context.parameters["keyId"] ?: throw IllegalArgumentException("No key id provided.")
 
                 val success = getWalletService().deleteKey(keyId)
+                context.respond(if (success) HttpStatusCode.Accepted else HttpStatusCode.BadRequest)
+            }
+
+            delete("remove", {
+                summary = "Remove a specific key (soft delete)"
+                response {
+                    HttpStatusCode.Accepted to { description = "Key removed" }
+                    HttpStatusCode.BadRequest to { description = "Failed to remove the key" }
+                }
+            }) {
+                val keyId = context.parameters["keyId"] ?: return@delete context.respond(
```

Comment:
this could be done also using [getOrFail](https://api.ktor.io/older/1.6.8/ktor-server/ktor-server-core/io.ktor.util/get-or-fail.html):
```kotlin
val keyId = context.parameters.getOrFail("keyId")
```
it throws a [MissingRequestParameterException](https://api.ktor.io/older/1.6.8/ktor-server/ktor-server-core/io.ktor.features/-missing-request-parameter-exception/index.html) that constructs the message similar to 'missing x parameter' and is also a BadRequestException, so it will be handled by service-commons